packages/middleware-endpoint-discovery/src/updateDiscoveredEndpointInCache.ts

```diff
@@ -0,0 +1,88 @@
+import { EndpointDiscoveryMiddlewareConfig } from "./getEndpointDiscoveryPlugin";
+import { EndpointDiscoveryResolvedConfig, PreviouslyResolved } from "./resolveEndpointDiscoveryConfig";
+
+export interface UpdateDiscoveredEndpointInCacheOptions extends EndpointDiscoveryMiddlewareConfig {
+  cacheKey: string;
+  commandName: string;
+  endpointDiscoveryCommandCtor: new (comandConfig: any) => any;
+}
+
+const requestQueue: { [key: string]: { resolve: Function; reject: Function }[] } = {};
```

Comment:
Nice work on this observer pattern implementation!

Reply:
Thank you for the suggestion, and reference implementation using callbacks in JS SDK v2 in https://github.com/aws/aws-sdk-js/blob/1a54a0993370cc2a22476df82142efbcec4b89b0/lib/discover_endpoint.js#L169-L197